Finish Darg'nDrop

StrongerProgrammer7 opened the pull request:
Add all method for move elements to box 
Box left - flex div 
Box right - absolute

StrongerProgrammer7 committed:
Added folder assets and create file
Added index html and
created container

StrongerProgrammer7 committed:
Created structure project
Added imgs for elements
Added classes and reset

StrongerProgrammer7 committed:
Added name every boxes

StrongerProgrammer7 committed:
Added style for img and box
added event:
DOMContentLoaded =>
initial elements for move
pointMove
pointerup => for put to box
Added predicat
isMoveElem
isCurrentBox
Finish Box flex
TODO: Box freedom

StrongerProgrammer7 committed:
Added the ability listener wheel

StrongerProgrammer7 committed:
Finish save elem to box freedom
Added save coordinate in box
Fix problem with start show new elem
Added event listener for box freedom for save coordinate

StrongerProgrammer7 committed:
Added extra div for color box=>
mouse move
Added create beautifulWord
Added methods for managment change color bakcground
Fix some problem with move mouse
Added snow

StrongerProgrammer7 committed:
change color body background
Change place div block background for box

StrongerProgrammer7 committed:
Add gh-pages

StrongerProgrammer7 committed:
cursor movement handler - work after pointerdown and
remove after pointerup

StrongerProgrammer7 committed:
Added cache of images capability

StrongerProgrammer7 committed:
Made placement img:
pixel in pixel

brianconnoly commented on assets/js/script.js:
```diff
@@ -0,0 +1,350 @@
+
+if(boxFreedom)
+{
+    boxFreedom.addEventListener('pointermove',(e) =>
+    {
+        if(isMoveElem(absCopyNode))
+        {
+            const rect = boxFreedom.getBoundingClientRect();
+
+            coordinateByDivFreedom[0] = e.clientX - rect.left - coordinateMouseOnImg_X;//(absCopyNode.width / 2); - if center of cursor on mouse
+            coordinateByDivFreedom[1] = e.clientY - rect.top - coordinateMouseOnImg_Y;//(absCopyNode.height / 2 ); - if center of cursor on mouse
+        
+        }
+    });
+}
```
Лучше одним событием `pointermove` на документе обойтись, используй target/currentTarget

StrongerProgrammer7 replied:
Fix!

brianconnoly commented on assets/js/script.js:
```diff
@@ -0,0 +1,350 @@
+    });
+}
+
+document.addEventListener('pointerup', (e) =>
```
тоже лучше назначать обработчик в начале перетаскивания и снимать при отпускании

StrongerProgrammer7 replied:
Fix!

StrongerProgrammer7 committed:
Fix:
To the document only one event listener: pointermove
The event "pointerup" set in begin move and delete after put
